lib/workers/types.ts

```diff
@@ -127,6 +130,173 @@ export interface BranchConfig
   isConflicted?: boolean;
 }
 
+export interface NarrowedBranchConfig extends Record<string, any> {
```

Comment:
@zharinov looks the wrong way to me.

 i think duplicate all those properties makes further null checks more complicated. also maintainability will be more worse.

Reply:
We don't need to implement narrowing. I created the narrowed config only to make it easy to track fields that are necessary for caching.